### PR TITLE
Python3 Fixes

### PR DESCRIPTION
**Pull request by GDYendell**
Fixes to some issues not directly related to, but found during the process updating to use python3 at DLS.

This is based on #45 because I didn't want to deal with rebasing them, so that will have to go in first. Only the top four commits here are new, starting from [Fix event loop error](https://github.com/dls-controls/excalibur-detector/pull/46/commits/bbb9554abd2c8250c5975378428af80b17a3cd8c).

- [x] #45

### CODE REVIEW

**Comment by GDYendell** [comment]
I would also like to add GitHub actions to confirm the build works outside DLS.

File: python/src/excalibur_detector/control/hl_detector.py [hunk 1]
```diff
@@ -1524,7 +1524,8 @@ def power_card_read(self):
             # Check for the current HV enabled state
             hv_enabled = 0
             # Greater than hv_bias means the HV is enabled
-            if self._powercard_status['pwr_bias_vmon'] > self._hv_bias - 5.0:
```
Comment (ajgdls) [inline python/src/excalibur_detector/control/hl_detector.py:1527]: Not part of this fix but should we fix that magic number (5.0)?

**Review by ajgdls — APPROVED** [review]

File: python/src/excalibur_detector/control/hl_detector.py (hunk 1 above)
Comment (GDYendell) [reply python/src/excalibur_detector/control/hl_detector.py:1527]: Is [this](https://github.com/dls-controls/excalibur-detector/blob/packaging/python/src/excalibur_detector/control/hl_detector.py#L1460) the same 5.0 or a different 5.0?

**Comment by GDYendell** [comment]
I have removed the change to the powercard params to fix an error when running without a detector because
1. I don't know what effect it will have on the other powercard params that are actually lists
2. The hl adapter has a simulator mode and the error doesn't happen in that case

For information, this is the error when running without a detector and simluated = False:
```
Traceback (most recent call last):
  ...
  File ".../excalibur_detector/control/hl_detector.py", line 1527, in power_card_read
    if self._powercard_status['pwr_bias_vmon'] > self._hv_bias - 5.0:
TypeError: '>' not supported between instances of 'list' and 'float'
```

**Comment by GDYendell** [comment]
Ready for review @timcnicholls

**Review by timcnicholls — APPROVED** [review]
Not in a position to test these changes in any meaningful way but they look sane so happy to approve

**Pull request merged**